lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,714 @@
+  let accessKeyId = username;
+  let secretAccessKey = password;
+  let region;
+
+  if (!accessKeyId) {
+    accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  }
+  if (!secretAccessKey) {
+    secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  }
+  if (endpoint) {
+    const regionReg = regEx(/.*codecommit\.(?<region>.+)\.amazonaws\.com/);
+    const codeCommitMatch = regionReg.exec(endpoint);
+    region = codeCommitMatch?.groups?.region;
+    if (!region) {
+      logger.warn("Can't parse region, make sure your endpoint is correct");
+    }
+  } else {
+    region = process.env.AWS_REGION;
+  }
+
+  if (!accessKeyId || !secretAccessKey || !region) {
+    throw new Error(
+      'Init: You must configure a AWS user(accessKeyId), password(secretAccessKey) and endpoint/AWS_REGION'
+    );
+  }
+
+  config.region = region;
+  const credentials = {
+    accessKeyId,
+    secretAccessKey,
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  };
+  config.credentials = credentials;
+
+  initIamClient(region, credentials);
+  config.userArn = await getUserArn();
+
+  client.buildCodeCommitClient(region, credentials);
+
+  const platformConfig: PlatformResult = {
+    endpoint: endpoint ?? `https://git-codecommit.${region}.amazonaws.com/`,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+export async function initRepo({
+  repository,
+  endpoint,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`initRepo("${repository}")`);
+
+  config.repository = repository;
+  const url = getCodeCommitUrl(config.region!, repository, config.credentials!);
+  try {
+    await git.initRepo({
+      url,
+    });
+  } catch (err) {
+    logger.debug({ err }, 'Failed to git init');
+    throw new Error(PLATFORM_BAD_CREDENTIALS);
+  }
+
+  let repo;
+  try {
+    repo = await client.getRepositoryInfo(repository);
+  } catch (err) {
+    logger.error({ err }, 'Could not find repository');
+    throw new Error(REPOSITORY_NOT_FOUND);
+  }
+
+  if (!repo) {
+    logger.error({ repository }, 'Could not find repository');
+    throw new Error(REPOSITORY_NOT_FOUND);
+  }
+
+  logger.debug({ repositoryDetails: repo }, 'Repository details');
+  const metadata = repo.repositoryMetadata;
+
+  if (!metadata?.defaultBranch || !metadata?.repositoryId) {
+    logger.debug('Repo is empty');
+    throw new Error(REPOSITORY_EMPTY);
+  }
+
+  const defaultBranch = metadata.defaultBranch;
+  config.defaultBranch = defaultBranch;
+  logger.debug(`${repository} default branch = ${defaultBranch}`);
+
+  return {
+    repoFingerprint: repoFingerprint(metadata.repositoryId, endpoint),
+    defaultBranch,
+    isFork: false,
+  };
+}
+
+export async function getPrList(): Promise<Pr[]> {
+  logger.debug('getPrList()');
+
+  const listPrsResponse = await client.listPullRequests(
+    config.repository!,
+    config.userArn!
+  );
+  const fetchedPrs: Pr[] = [];
+
+  /* istanbul ignore if */
+  if (listPrsResponse && !listPrsResponse.pullRequestIds) {
+    return fetchedPrs;
+  }
```

Comment:
can't the code commit client be mocked?

Reply:
basically they are never empty, so code shouldn't reach inside here Anytime, i'm just avoiding null checks.

Reply:
done